pandas/io/formats/format.py

```diff
@@ -1527,6 +1528,8 @@ def _format_strings(self) -> list[str]:
         if isinstance(values, Categorical):
             # Categorical is special for now, so that we can preserve tzinfo
             array = values._internal_get_values()
+        elif isinstance(values, BaseMaskedArray):
+            array = values.to_numpy(na_value=NA)
```

Comment:
As I discovered in https://github.com/pandas-dev/pandas/pull/56003/files#diff-c2df0eefbdda16cb85d5e0a305db90101f92bfb790a8e989ab2e99b935b7f6b3, maybe we just want to do `to_numpy(dtype=object)` here?